cleanup kernel options

xaverdh opened the pull request:
###### Motivation for this change
I hope I did the right thing with the old options, please check. Not quite sure what `CIFS_WEAK_PW_HASH` is about anyway..


###### Things done

* ~~I built the testing kernel.~~
* ~~My system is busy recompiling~~ recompiled all kernels in `nixpkgs`, to see if they still build.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

xaverdh commented:
cc @TredwellGit @alyssais

Artturin commented:
@ofborg eval

xaverdh commented:
dropped the commit that updates the testing kernel, this pr now does cleanup only.

klemensn commented:
> I hope I did the right thing with the old options, please check. Not quite sure what `CIFS_WEAK_PW_HASH` is about anyway..

https://github.com/NixOS/nixpkgs/pull/164906 is a partial duplicate with proper references as to why respective options should be removed/set conditionally.

xaverdh commented:
> > I hope I did the right thing with the old options, please check. Not quite sure what `CIFS_WEAK_PW_HASH` is about anyway..
> 
> #164906 is a partial duplicate with proper references as to why respective options should be removed/set conditionally.

well technically I do have proper references, but they are in the commit messages.. but either way I am fine with focusing on the other pr as well

klemensn commented:
> > > I hope I did the right thing with the old options, please check. Not quite sure what `CIFS_WEAK_PW_HASH` is about anyway..
> > 
> > 
> > #164906 is a partial duplicate with proper references as to why respective options should be removed/set conditionally.
> 
> well technically I do have proper references, but they are in the commit messages.. but either way I am fine with focusing on the other pr as well

Great, that's basically what I asked for in the other PR.

fabianhjr commented on pkgs/os-specific/linux/kernel/common-config.nix:
```diff
@@ -597,10 +597,11 @@ let
-      XEN_SELFBALLOONING          = option yes;
-      XEN_STUB                    = option yes;
-      XEN_TMEM                    = option yes;
+      XEN_SCRUB_PAGES             = { optional = true; tristate = whenOlder "4.19" "y"; };
+      XEN_SCRUB_PAGES_DEFAULT     = { optional = true; tristate = whenAtLeast "4.19" "y"; };
+      XEN_SELFBALLOONING          = { optional = true; tristate = whenOlder "5.3" "y"; };
+      XEN_STUB                    = { optional = true; tristate = whenOlder "5.13" "y"; };
+      XEN_TMEM                    = { optional = true; tristate = whenOlder "5.3" "y"; };
```
why not `whenOlder x (option yes)`?

Atemu commented:
@xaverdh could you rebase, remove special handling for kernels <4.14 and use the `option` function rather than declaring the tristates manually?

xaverdh committed:
linux-kernel: clean up XEN related options in newer kernels

Upstream commit 197ecb3802c04499d8ff4f8cb28f6efa008067db renamed XEN_SCRUB_PAGES into XEN_SCRUB_PAGES_DEFAULT.
Also since 01325044dbe47a7dc66b786445727a6de304f328 XEN_STUB is gone and
in 814bbf49dcd0ad642e7ceb8991e57555c5472cce the XEN_TMEM and related XEN_SELFBALLOONING option were dropped.

Atemu commented on pkgs/os-specific/linux/kernel/common-config.nix:
```diff
@@ -632,6 +632,9 @@ let
       XEN_SCRUB_PAGES             = whenOlder "4.19" yes;
       XEN_SELFBALLOONING          = whenOlder "5.3" yes;
+      XEN_SCRUB_PAGES_DEFAULT     = whenAtLeast "4.19" (option yes);
+      XEN_STUB                    = whenOlder "5.13" (option yes);
+      XEN_TMEM                    = whenOlder "5.3" (option yes);
```
These two are probable not even desirable when they were a thing.

Atemu reviewed (approved):
I think we should just remove STUB and TMEM entirely but otherwise LGTM.

Nyabinary commented:
This pr is stale correct?

xaverdh commented:
Yes, will close
